packages/lesswrong/unitTests/sql/FragmentLexer.tests.ts

```diff
@@ -0,0 +1,34 @@
+import FragmentLexer from "../../lib/sql/FragmentLexer";
+
+describe("FragmentLexer", () => {
+  it("can lex GraphQL fragments", () => {
+    const lexer = new FragmentLexer(`
+      fragment TestFragment on TestCollection {
+        _id
+        a # with a comment
+        ...SomeOtherFragment
+        b {
+          c
+          yetAnotherFragment
+        }
+      }
+    `);
+    expect(lexer.getName()).toBe("TestFragment");
+    expect(lexer.getBaseTypeName()).toBe("TestCollection");
+    expect(lexer.isFinished()).toBe(false);
+    expect(lexer.next()).toBe("_id");
+    expect(lexer.isFinished()).toBe(false);
+    expect(lexer.next()).toBe("a");
+    expect(lexer.isFinished()).toBe(false);
+    expect(lexer.next()).toBe("...SomeOtherFragment");
+    expect(lexer.isFinished()).toBe(false);
+    expect(lexer.next()).toBe("b {");
```

Comment:
I doesn't have to be like this - it just made it easier to write the regexes in the parser. I'm going to leave as-is for now since, as described above, I think we might switch to using the Apollo AST in the future.